feat: FIT-629: Create a reusable Empty State component

ricardoantoniocm opened the pull request:
This pull request introduces a new reusable `EmptyState` component to the UI library, designed for displaying empty or placeholder states across the application. The implementation includes the component itself, its SCSS module for styling, a comprehensive test suite, and updates to the UI library exports to make the component available for use. The component supports multiple sizes, color variants, accessibility features, and flexible content areas.

Note: Separate PRs will tackle replacing existing empty states.

**New EmptyState component:**

* Implemented the `EmptyState` React component with support for three sizes (`large`, `medium`, `small`), six color variants, customizable icon sizing, flexible content areas (actions, additional content, footer), and full accessibility support. (`empty-state.tsx`)
* Added a dedicated SCSS module for the `EmptyState` component, providing base styles, size-specific layouts, color variants, and a gradient theme with animation. (`empty-state.module.scss`)

**Screenshot**
<img width="1052" height="1384" alt="image" src="https://github.com/user-attachments/assets/c2d5636f-cfb2-4749-8e1e-4316c3ad4291" />


**Testing and exports:**

* Created a thorough test suite for the `EmptyState` component covering rendering, props, accessibility, and content flexibility. (`empty-state.spec.tsx`)
* Updated UI library exports to include the new `EmptyState` component and its types, making it available for use throughout the application. (`index.ts`, `lib/empty-state/index.ts`) [[1]](diffhunk://#diff-1b7b120e6194326932be999557506481a8f7991798c7c3ee3d6e4edc42ad41a4R13) [[2]](diffhunk://#diff-eadaed672261efeed5b40f2bcc5686afa9e22778cebd974d1d03368e74171305R1-R2)

ricardoantoniocm committed:
feat: FIT-629: Create a reusable Empty State component

ricardoantoniocm committed:
Implements Empty State component.

ricardoantoniocm committed:
Linter fixes.

ricardoantoniocm committed:
Implements fixes.

ricardoantoniocm committed:
Linter fixes.

ricardoantoniocm committed:
Linter fixes.

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/17276787968)
> Successfully merged:  create mode 100644 docs/themes/v2/source/images/org-id.png

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-629/empty-state'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17276787968

ricardoantoniocm committed:
Fixes size classes.

nass600 reviewed (approved):
Amazing PR 👏 , great Job!!

ricardoantoniocm committed:
Implements new empty-state component on Machine Learning and Prediction settings pages. Updates models icon to use currentColor.
Adjusts various style elements.

ricardoantoniocm commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/17299702347)

ricardoantoniocm committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17299702347

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-629/empty-state'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17299702347

hlomzik commented on web/libs/ui/src/lib/empty-state/empty-state.tsx:
```diff
@@ -0,0 +1,269 @@
+      className,
+      "data-testid": testId,
+      "aria-label": ariaLabel,
+      titleId,
+      descriptionId,
```
Could you please provide examples how to use these params? Feels like they are just local ids, so do we really need to expose them?

ricardoantoniocm replied:
The cases where these would be set manually would be exceptions, so I hid them as their values are auto-generated and that will work for 99.99% of cases. Thank you for pointing this out!

ricardoantoniocm committed:
Adds additional guidance for accessibility-related params.

ricardoantoniocm committed:
Hides accessibility params that don't need to be exposed.

hlomzik commented on web/libs/ui/src/lib/empty-state/empty-state.tsx:
```diff
@@ -0,0 +1,322 @@
+          <div className={cn("empty-state__actions", styles.actions)}>
+            {(() => {
+              // Flatten children and filter out null/false values to count actual rendered elements
+              const flattenedActions = React.Children.toArray(actions).flat().filter(Boolean);
```
it's already a flat array

```suggestion
              // Filter out null/false values to count actual rendered elements
              const realActions = React.Children.toArray(actions).filter(Boolean);
```

ricardoantoniocm committed:
Update web/libs/ui/src/lib/empty-state/empty-state.tsx

Co-authored-by: Andrew <hlomzik@gmail.com>

hlomzik commented on web/libs/ui/src/lib/empty-state/empty-state.tsx:
```diff
@@ -0,0 +1,322 @@
+            {(() => {
+              // Filter out null/false values to count actual rendered elements
+              const flattenedActions = React.Children.toArray(actions).filter(Boolean);
+              const actualActionCount = flattenedActions.length;
+              const isSingleAction = actualActionCount === 1;
```
```suggestion
              const isSingleAction = realActions.length === 1;
```

ricardoantoniocm committed:
Adjusts spacing between elements. Adds Relations example. Refactors action array based on feedback.

ricardoantoniocm committed:
Removes unnecessary intermediate variable based on feedback.

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/17412884002)
> Successfully merged:  create mode 100644 web/libs/editor/src/tools/Vector.js

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-629/empty-state'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17412884002

ricardoantoniocm committed:
Increases max-width of descriptions for medium-size variant.

ricardoantoniocm committed:
Improves copy of Model and Predictions pages.

ricardoantoniocm committed:
Improves copy of Model settings page.
